### PR TITLE
Did::parse & unit test coverage

### PR DESCRIPTION
**Pull request by KendallWeihe**
- Rename `Did::new()` to `Did::parse()`
- Use `Web5Error` inside of `did.rs`
- Replace `OnceCell` with latest `LazyLock` feature
- Write kotlin implementation for `Did.parse()`
- Add comprehensive unit test coverage

### CODE REVIEW

File: bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt [hunk 1]
```diff
@@ -1,22 +1,249 @@
...
+import org.junit.jupiter.api.*
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.fail
+import web5.sdk.UnitTestSuite
+import web5.sdk.rust.Web5Exception
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class DidTest {
 
+  private val testSuite = UnitTestSuite("did_new")
```
Comment (nitro-neal) [inline bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt:12]:
```suggestion
  private val testSuite = UnitTestSuite("did_parse")
```

Comment (KendallWeihe) [reply bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt:12]: ah yes, love it

File: bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt [hunk 2]
```diff
@@ -1,22 +1,249 @@
...
+      Did.parse(uri)
+    }
+
+    assertEquals("parameter error identifier regex match failure $uri", exception.msg)
+  }
+
+  @Test
+  fun test_did_missing_id_part_should_error() {
+    this.testSuite.include()
+    val uri = "did:uport"
```
Comment (nitro-neal) [inline bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt:50]: works, but kind of strange to be using uport

**Review by nitro-neal — APPROVED** [review]
awesome!

File: bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt (hunk 2 above)
Comment (KendallWeihe) [reply bound/kt/src/test/kotlin/web5/sdk/dids/DidTest.kt:50]: most of this is copy pasta from web5-go, so idunno what the origin of that is but meh it's fine right?

**Comment by KendallWeihe** [comment]
Merging this, but @diehuxx to your point, the `LazyLock` code will likely need to be redone if we revert our MVRV back to something more widely accepted

**Pull request merged**